nix/cabal-docspec.nix

```diff
@@ -0,0 +1,18 @@
+{ pkgs ? import <nixpkgs> {} }:
+
+pkgs.stdenv.mkDerivation {
+  name = "cabal-docspec";
+
+  src = pkgs.fetchurl {
+    url = "https://github.com/phadej/cabal-extras/releases/download/cabal-docspec-0.0.0.20211114/cabal-docspec-0.0.0.20211114.xz";
```

Comment:
Honestly, considering the constraints, this may be the best solution available to us, since it seems to be working well enough.

Reply:
I would think so. In the end, we are not redistributing nor depending on `cabal-docspec`. It's an utility anyone can use to run the doctests. Perhaps I could include some instructions about getting and using `cabal-docspec`, from a developer point of view.